pkg/reconciler/reconciler.go

```diff
@@ -0,0 +1,75 @@
+package reconciler
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type Reconciler interface {
+	Reconcile(ctx context.Context, c client.Client, scheme *runtime.Scheme) error
+}
+
+type UpdateReconciler struct {
```

Comment:
I think we can avoid stuttering by omit `Reconciler` e.g.`reconciler.NewUpdateReconciler` would read `reconciler.Updater`

Reply:
I'd agree with @sthaha that `reconciler.NewUpdateReconciler` is a bit verbose. `reconciler.Updater` and `reconciler.Deleter` sound good to me.
Not sure how to rename `NewOptionalResourceReconciler()` though. Maybe it can stay the same?